examples/digital_fingerprinting/production/morpheus/dfp/stages/multi_file_source.py

```diff
@@ -0,0 +1,127 @@
+
+    Parameters
+    ----------
+    c : `morpheus.config.Config`
+        Pipeline configuration instance.
+    filename : str
+        Name of the file from which the messages will be read.
+    iterative: boolean
+        Iterative mode will emit dataframes one at a time. Otherwise a list of dataframes is emitted. Iterative mode is
+        good for interleaving source stages.
+    file_type : `morpheus._lib.file_types.FileTypes`, default = 'auto'
+        Indicates what type of file to read. Specifying 'auto' will determine the file type from the extension.
+        Supported extensions: 'json', 'csv'
+    repeat: int, default = 1
+        Repeats the input dataset multiple times. Useful to extend small datasets for debugging.
+    filter_null: bool, default = True
+        Whether or not to filter rows with null 'data' column. Null values in the 'data' column can cause issues down
+        the line with processing. Setting this to True is recommended.
+    cudf_kwargs: dict, default=None
+        keyword args passed to underlying cuDF I/O function. See the cuDF documentation for `cudf.read_csv()` and
+        `cudf.read_json()` for the available options. With `file_type` == 'json', this defaults to ``{ "lines": True }``
+        and with `file_type` == 'csv', this defaults to ``{}``.
+    """
+
+    def __init__(
+        self,
+        c: Config,
+        filenames: typing.List[str],
+    ):
+        super().__init__(c)
+
+        self._batch_size = c.pipeline_batch_size
+
+        self._filenames = filenames
+
+        self._input_count = None
+        self._max_concurrent = c.num_threads
+
+    @property
+    def name(self) -> str:
+        return "from-multi-file"
+
+    @property
+    def input_count(self) -> int:
+        """Return None for no max intput count"""
+        return self._input_count
+
+    def supports_cpp_node(self):
+        return False
+
+    def _generate_frames_fsspec(self):
+
+        files: fsspec.core.OpenFiles = fsspec.open_files(self._filenames, filecache={'cache_storage': './.cache/s3tmp'})
+
+        if (len(files) == 0):
+            raise RuntimeError(f"No files matched input strings: '{self._filenames}'. "
+                               "Check your input pattern and ensure any credentials are correct")
+
+        yield files
+
+    def _generate_frames(self):
+
+        loaded_dfs = []
+
+        for f in self._filenames:
+
+            # Read the dataframe into memory
+            df = read_file_to_df(f,
+                                 self._file_type,
+                                 filter_nulls=True,
+                                 df_type="pandas",
+                                 parser_kwargs=self._parser_kwargs)
+
+            df = process_dataframe(df, self._input_schema)
+
+            loaded_dfs.append(df)
+
+        combined_df = pd.concat(loaded_dfs)
+
+        print("Sending {} rows".format(len(combined_df)))
+
+        yield combined_df
```

Comment:
I don't think this method is in use anymore

Reply:
Done

Reply:
Removed